esp/src/src-react/components/forms/landing-zone/BlobImportForm.tsx

```diff
@@ -0,0 +1,384 @@
+                        }}
+                        errorMessage={ error && error.message }
+                    /> }
+                    rules={{
+                        required: `Select a ${nlsHPCC.Queue}`
+                    }}
+                />
+                <Controller
+                    control={control} name="destLogicalName"
+                    render={({
+                        field: { onChange, name: fieldName, value },
+                        fieldState: { error }
+                    }) => <TextField
+                        name={fieldName}
+                        onChange={onChange}
+                        label={nlsHPCC.TargetScope}
+                        value={value}
+                        placeholder={nlsHPCC.TargetNamePlaceholder}
+                        errorMessage={ error && error.message }
+                    /> }
+                    rules={{
+                        pattern: {
+                            value: /^([a-z0-9]+(::)?)+$/i,
+                            message: nlsHPCC.ValidationErrorNamePrefix
+                        }
+                    }}
+                />
+            </Stack>
+            <Stack>
+                <table className={`${componentStyles.twoColumnTable} ${componentStyles.selectionTable}`}>
+                    <thead>
+                        <tr>
+                            <th>{nlsHPCC.SourcePath}</th>
+                        </tr>
+                    </thead>
+                    <tbody>
+                    { selection.length && selection.map((file, idx) => {
+                        return <tr key={`File-${idx}`}>
+                            <td>
+                                <Controller
+                                    control={control} name={`selectedFiles.${idx}.SourceFile` as const}
+                                    render={({
+                                        field: { onChange, name: fieldName, value },
+                                        fieldState: { error }
+                                    }) => <TextField
+                                        name={fieldName}
+                                        onChange={onChange}
+                                        value={value}
+                                        errorMessage={ error && error.message }
+                                    /> }
+                                    rules={{
+                                        required: nlsHPCC.ValidationErrorTargetNameRequired,
+                                        pattern: {
+                                            value: /^([a-z0-9]+[-a-z0-9 \._]+)+$/i,
+                                            message: nlsHPCC.ValidationErrorTargetNameInvalid
+                                        }
+                                    }}
+                                />
+                                <input type="hidden" { ...register(`selectedFiles.${idx}.SourceIP` as const) } value={file["NetAddress"]} />
```

Comment:
Expensive, foces re-refresh every time...  Looks like it shoould be a "useMemo"?